src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryCustomObject.cs

```diff
@@ -0,0 +1,250 @@
+            }
+
+            public ICustomObject this[int index] => new JsonCustomObject(_element[index]);
+
+            public bool IsArray => _element.ValueKind == JsonValueKind.Array;
+
+            public bool IsNull => _element.ValueKind == JsonValueKind.Null;
+
+            public bool IsObject => _element.ValueKind == JsonValueKind.Object;
+
+            public bool IsString => _element.ValueKind == JsonValueKind.String;
+
+            public bool IsNumber => _element.ValueKind == JsonValueKind.Number;
+
+            public int GetArrayLength()
+            {
+                return _element.GetArrayLength();
+            }
+
+            public double GetDouble()
+            {
+                return _element.GetDouble();
+            }
+
+            public string GetString()
+            {
+                return _element.GetString();
+            }
+
+            public object ToObject()
+            {
+                return _element.GetRawText();
+            }
+
+            public bool TryGetProperty(string value, out ICustomObject result)
+            {
+                JsonElement je;
+                var res = _element.TryGetProperty(value, out je);
+                result = new JsonCustomObject(je);
+                return res;
+            }
+        }
+
+        public static FormulaValue ParseJson(IRContext irContext, StringValue[] args)
+        {
+            var json = args[0].Value;
+            JsonElement result;
+
+            try
+            {
+                using (JsonDocument document = JsonDocument.Parse(json))
+                {
+                    // Clone must be used here because the original element will be disposed
+                    result = document.RootElement.Clone();
+                }
+
+                // Map null to blank
+                if (result.ValueKind == JsonValueKind.Null)
+                {
+                    return new BlankValue(IRContext.NotInSource(FormulaType.Blank));
+                }
+
+                return new CustomObjectValue(irContext, new JsonCustomObject(result));
+            }
+            catch (JsonException ex)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = $"The Json could not be parsed: {ex.Message}",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.InvalidFunctionUsage
+                });
+            }
+        }
+
+        public static FormulaValue GetAt(IRContext irContext, FormulaValue[] args)
+        {
+            var arg0 = (CustomObjectValue)args[0];
+            var arg1 = (NumberValue)args[1];
+
+            var element = arg0.Impl;
+
+            if (element.IsArray)
+            {
+                int len = element.GetArrayLength();
+                var index = (int)arg1.Value;
+
+                if (index < len)
+                {
+                    ICustomObject result = element[index];
```

Comment:
PowerFx is intended to be 1-indexed (@gregli-msft to double check).

Reply:
Fixed